### ISSUE TITLE
Controllers are not deleted after the route is removed by offAllNamed

### ISSUE THREAD

**Issue by csnguyen-gmail** [jonataslaw/getx#1748]
**Describe the bug**
I bind 2 controllers to a route, after I switch to another route with Get.offAllNamed function, only the last controller instance was deleted. I suppose all controllers are deleted.

**Reproduction code

```dart
import 'package:flutter/material.dart';
import 'package:get/get.dart';

void main() {
  runApp(GetMaterialApp(
    initialRoute: '/',
    getPages: [
      GetPage(
        name: '/',
        page: () => HomePage(),
        binding: BindingsBuilder(() {
          Get.put(HomeController());
        }),
      ),
      GetPage(
        name: '/detail',
        page: () => DetailPage(),
        binding: BindingsBuilder(() {
          Get.put(DetailController());
          Get.put(OtherController());
        }),
      ),
    ],
  ));
}

class HomeController extends GetxController {}

class DetailController extends GetxController {}

class OtherController extends GetxController {}

class HomePage extends GetView<HomeController> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("Home"),
      ),
      body: Center(
        child: TextButton(
          onPressed: () {
            Get.offAllNamed('/detail');
          },
          child: Text("Off all to Detail"),
        ),
      ),
    );
  }
}

class DetailPage extends GetView<DetailController> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("Detail"),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            TextButton(
              onPressed: () {
                Get.offAllNamed('/');
              },
              child: Text("Off all to Home"),
            ),
          ],
        ),
      ),
    );
  }
}
```

**To Reproduce**
1. On Home page tap to "Off all to Detail" to off to Detail page
2. On Detail page tap to "Off all to Home" to off to Home page
3. See the log

**Expected behavior**
DetailController should be deleted

**Screenshots**
![image](https://user-images.githubusercontent.com/3714474/128759632-429aefbe-e6e8-44a4-b1cf-818d07056911.png)

**Flutter Version:**
Flutter (Channel stable, 2.2.3, on macOS 11.5 20G71 darwin-arm, locale en-VN)

**Getx Version:**
get 4.3.5

**Comment by jonataslaw** [comment]
Thanks for the reproduce code, this will be fixed asap.

**Comment by jonataslaw** [comment]
Fixed in get 4.3.6
